xmake/rules/c++/modules/build_modules/clang.lua

```diff
@@ -100,6 +100,14 @@ function build_with_batchjobs(target, batchjobs, sourcebatch, opt)
             opt2.objectfile = modulefiles[i]
             opt2.dependfile = target:dependfile(opt2.objectfile)
             opt2.sourcekind = assert(sourcebatch.sourcekind, "%s: sourcekind not found!", sourcefile)
+            if moduledep["deps"] then
+                for _, modulefile in ipairs(modulefiles) do
+                    local found = modulefile:find(moduledep["name"])
+                    if found == nil then
+                        target:add("cxxflags", "-fmodule-file=" .. modulefile, {force = true})
+                    end
+                end
```

Comment:
why? The code below will add it

Reply:
https://github.com/xmake-io/xmake/blob/1b2a5da03aea8f9082204c345e852ee69bd6cec4/xmake/rules/c%2B%2B/modules/build_modules/clang.lua#L122-L129

Reply:
The code above doesn't add the pcm file to the compilation.

Reply:
We've turned on -fimplicit-modules and module cache, clang should find submodules without explicitly setting -fmodule-file.

The root cause of it is that the module files for cache/module1.pcm are not named correctly, in case of submodules, it should be cache/main_module.module1.pcm so that clang can find them correctly.

I have tested your demo project and confirmed that it can be fixed successfully.